consensus/src/transaction_shuffler/fairness/conflict_key/mod.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction_shuffler::fairness::{TxnIdx, NUM_CONFLICT_ZONES};
+use aptos_types::transaction::SignedTransaction;
+use entry_fun::EntryFunKey;
+use entry_fun_module::EntryFunModuleKey;
+use std::{collections::HashMap, hash::Hash};
+use txn_sender::TxnSenderKey;
+
+mod entry_fun;
+mod entry_fun_module;
+mod txn_sender;
+
+#[cfg(test)]
+mod test_utils;
+
+pub(crate) trait ConflictKey: Eq + Hash + PartialEq {
+    fn extract_from(txn: &SignedTransaction) -> Self;
+
+    fn conflict_exempt(&self) -> bool {
+        false
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct ConflictKeyId(usize);
+
+impl ConflictKeyId {
+    pub fn as_idx(&self) -> usize {
+        self.0
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ConflictKeyRegistry {
+    id_by_txn: Vec<ConflictKeyId>,
```

Comment:
Some comments here explaining these data structures would be helpful.

Reply:
I'm a believer of Code As Documentation, but okey..